python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -184,9 +199,18 @@ def required_resources(self) -> Resources:
             self.placement_group_bundles is not None
             and self.placement_group_strategy == "STRICT_PACK"
         ):
-            return sum(self.placement_group_bundles, Resources())
+            required = sum(self.placement_group_bundles, Resources())
         else:
-            return self.actor_resources
+            required = self.actor_resources
+
+        if self.max_replicas_per_node:
+            implicit_resource = (
+                f"{ray._raylet.IMPLICIT_RESOURCE_PREFIX}"
+                f"{self.deployment_id.app_name}:{self.deployment_id.name}"
+            )
+            required[implicit_resource] = 1.0 / self.max_replicas_per_node
```

Comment:
Move this into a helper function so the code can be reused.

Reply:
I will make a follow up PR to unify the two data structures, if that's OK.

Reply:
sure